aws/resource_aws_s3control_bucket_lifecycle_configuration_test.go

```diff
@@ -0,0 +1,750 @@
+			return fmt.Errorf("error parsing S3 Control Bucket ARN (%s): %w", rs.Primary.ID, err)
+		}
+
+		input := &s3control.GetBucketLifecycleConfigurationInput{
+			AccountId: aws.String(parsedArn.AccountID),
+			Bucket:    aws.String(rs.Primary.ID),
+		}
+
+		_, err = conn.GetBucketLifecycleConfiguration(input)
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_AbortIncompleteMultipartUpload_DaysAfterInitiation(rName string, daysAfterInitiation int) string {
+	return fmt.Sprintf(`
+data "aws_outposts_outposts" "test" {}
+
+data "aws_outposts_outpost" "test" {
+  id = tolist(data.aws_outposts_outposts.test.ids)[0]
+}
+
+resource "aws_s3control_bucket" "test" {
+  bucket     = %[1]q
+  outpost_id = data.aws_outposts_outpost.test.id
+}
+
+resource "aws_s3control_bucket_lifecycle_configuration" "test" {
+  bucket = aws_s3control_bucket.test.arn
+
+  rule {
+    abort_incomplete_multipart_upload {
+      days_after_initiation = %[2]d
+    }
+
+    expiration {
+      days = 365
+    }
+
+    id = "test"
+  }
+}
+`, rName, daysAfterInitiation)
+}
+
+func testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_Date(rName string, date string) string {
+	return fmt.Sprintf(`
+data "aws_outposts_outposts" "test" {}
+
+data "aws_outposts_outpost" "test" {
+  id = tolist(data.aws_outposts_outposts.test.ids)[0]
+}
+
+resource "aws_s3control_bucket" "test" {
+  bucket     = %[1]q
+  outpost_id = data.aws_outposts_outpost.test.id
+}
+
+resource "aws_s3control_bucket_lifecycle_configuration" "test" {
+  bucket = aws_s3control_bucket.test.arn
+
+  rule {
+    expiration {
+      date = %[2]q
+    }
+
+    id = "test"
+  }
+}
+`, rName, date)
+}
+
+func testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_Days(rName string, days int) string {
+	return fmt.Sprintf(`
+data "aws_outposts_outposts" "test" {}
+
+data "aws_outposts_outpost" "test" {
+  id = tolist(data.aws_outposts_outposts.test.ids)[0]
+}
+
+resource "aws_s3control_bucket" "test" {
+  bucket     = %[1]q
+  outpost_id = data.aws_outposts_outpost.test.id
+}
+
+resource "aws_s3control_bucket_lifecycle_configuration" "test" {
+  bucket = aws_s3control_bucket.test.arn
+
+  rule {
+    expiration {
+      days = %[2]d
+    }
+
+    id = "test"
+  }
+}
+`, rName, days)
+}
+
+func testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_ExpiredObjectDeleteMarker(rName string, expiredObjectDeleteMarker bool) string {
+	return fmt.Sprintf(`
+data "aws_outposts_outposts" "test" {}
+
+data "aws_outposts_outpost" "test" {
+  id = tolist(data.aws_outposts_outposts.test.ids)[0]
+}
+
+resource "aws_s3control_bucket" "test" {
+  bucket     = %[1]q
+  outpost_id = data.aws_outposts_outpost.test.id
+}
+
+resource "aws_s3control_bucket_lifecycle_configuration" "test" {
+  bucket = aws_s3control_bucket.test.arn
+
+  rule {
+    expiration {
+      days                         = %[2]t ? null : 365
```

Comment:
i think this line is causing the terrafmt error: `./aws/resource_aws_s3control_bucket_lifecycle_configuration_test.go:571 failed to process with: failed to parse hcl: ./aws/resource_aws_s3control_bucket_lifecycle_configuration_test.go:17,38-39: Invalid expression; Expected the start of an expression, but found an invalid expression token."`

Reply:
maybe `terrafmt` still needs to support ternaries w/formatters?

Reply:
We recently bumped `terrafmt` dependency on master and rebasing this pull request to fix the merge conflict fixed this. 👍